### ISSUE TITLE
Loss normalization in Video Tokenizer

### ISSUE THREAD

**Issue by witherhoard99** [nicklashansen/dreamer4#3]
Thank you for this implementation!

I was wondering why the Video Tokenizer does not have its losses normalized. The paper mentions [here](https://arxiv.org/pdf/2509.24527#page=5) that specifically video tokenizer losses are normalized ("To simplify weighing the two loss terms, we employ loss normalization as explained later."). 

I'm not sure where the explain later, it does seem like they use loss normalization. Was this choice from empirical results?

**Comment by nicklashansen** [comment]
Hi @witherhoard99, thanks for your interest! I missed this but also had another look at the paper and I could not find a description of their loss normalization for the tokenizer specifically, but they have this sentence describing a loss normalization for the dynamics model:

> To train a
single dynamics transformer with multiple modalities
and output heads, we normalize all loss terms by run-
ning estimates of their root-mean-square (RMS).

It would be interesting to see whether applying something like this to the tokenizer helps! My implementation is not a 1:1 reproduction of their work and I haven't really ablated any of the design decisions so I'm certain there's room for improvement here and there :-)

**Comment by witherhoard99** [comment]
It's very interesting that the tokenizer is able to work without loss normalization! LPIPS loss becomes weighted much heavier without loss normalization. I assumed that the authors used RMSNorm for the tokenizer as well, although this absolutely could be wrong.

When I was experimenting with the `lucidrains/dreamer4` repo, RMSNorm loss normalization for the Video Tokenizer was able to take it from non-convergence to able to showing some results. 

Would you say this overall dreamer 4 architecture is sensitive to hyper parameters?

**Comment by nicklashansen** [comment]
Interesting! I wasn't aware of the lucidrains repo actually so I'll have to dig more into the differences! I only trained a few models during development of this codebase so I don't believe it is particularly sensitive to hyper-parameters (or I got lucky) but I also didn't implement the full training pipeline proposed in the paper

**Comment by witherhoard99** [comment]
I will say that the lucidrains implementation has lots of things not in the paper like rnn_time, and muon optimizer.
